gmso/core/topology.py

```diff
@@ -1425,3 +1425,46 @@ def load(cls, filename, **kwargs):
 
         loader = LoadersRegistry.get_callable(filename.suffix)
         return loader(filename, **kwargs)
+
+    def convert_potential_styles(self, expressionMap={}):
+        """Convert from one parameter form to another.
+        Parameters
+        ----------
+        expressionMap : dict, default={}
+            map with keys of the potential type and the potential to change to
+
+        Examples
+        ________
+        # Convert from RB torsions to OPLS torsions
+        top.convert_potential_styles({"dihedrals": "OPLSTorsionPotential"})
+        """
+        from gmso.utils.conversions import convert_topology_expressions
+
+        return convert_topology_expressions(self, expressionMap)
+
+    def convert_unit_styles(self, unitsystem, exp_unitsDict):
+        """Convert from one set of base units to another.
+        Parameters
+        ----------
+        unitsystem : unyt.UnitSystem
+            set of base units to use for all expressions of the topology in unyt package
+        exp_unitsDict : dict
+            keys with topology attributes that should be converted and values with dictionary of parameter: expected_dimension
+
+        Examples
+        ________
+        # TODO
+        """
+        from gmso.utils.conversions import _convert_params_units
+
+        ref_values = {"energy": "kJ/mol", "length": "nm", "angle": "radians"}
+
+        # all potContainer ["atom", "bond", "angle", "dihedral", "improper"]
+        for potStr in exp_unitsDict:
+            potContainer = getattr(self, potStr + "_types")
```

Comment:
I think the initial goal for this method was just to be used on the parameters of the potential expressions, which is why it's missing things. Pairpotentials come to mind as an example of that. However, I ended up converting the expressions on writing out, instead of upfront in the lammpswriter, and mostly relying on methods that are living in lammpsdata.py because of that. I still think the best way to do the conversion would be:
1. Grab the set of base UnitSystem. Presumably, this would be from a library, or passed directly to whatever function by the user. I can see a useful added input of also passing units specific to parameters in the topology, such as: {"angle_types":{"theta":u.degrees}}, which would override the value for that UnitSystem.
2. Determine if the unit changes should happen in place or to a copy of the topology. This would be mostly for speedups, and prevent issues with overwriting whatever you pass to the function.
3. Iterate through all components of the topology. Each component should be passed to a generic unit_conversion method with the base units and the conversion should be done. These components include: [box, sites, atomtypes, bondtypes, angletypes, dihedraltypes, impropertypes, pairpotentials]. Each of these may have their own specific methods, but should be accessible through a single api.
4. Once the conversions are made, then the topology attributes can be written out directly as floats with the to_value() method. Rounding should also occur on writing.